src/document_basic.c

```diff
@@ -458,6 +460,7 @@ int Redis_SaveDocument(RedisSearchCtx *ctx, const AddDocumentOptions *opts, Quer
     RedisSearchCtx_LockSpecWrite(ctx);
     if (ctx->spec->rule->payload_field == NULL) {
       ctx->spec->rule->payload_field = rm_strndup(UNDERSCORE_PAYLOAD, strlen(UNDERSCORE_PAYLOAD));
+      IndexSpec_UpdateVersion(ctx->spec);
```

Comment:
I think we should change how we lock/unlock here and the version updates
there can be a case where all three conditions may apply